core/src/main/kotlin/org/evomaster/core/problem/rest/classifier/ModelAccuracyWithTimeWindow.kt

```diff
@@ -19,7 +22,29 @@ class ModelAccuracyWithTimeWindow(
         return ok / n
     }
 
-    override fun updatePerformance(predictionWasCorrect: Boolean) {
+    /**
+     * Precision for HTTP 400 responses: TP / (TP + FP).
+     */
+    override fun estimatePrecision400(): Double {
+        val tp = truePositive400Queue.count { it }   // count only true entries
+        val fp = falsePositive400Queue.count { it }  // count only true entries
+        return if ((tp + fp) > 0) tp.toDouble() / (tp + fp) else 0.0
+    }
```

Comment:
we need some basic unit tests to validate these computations (eg in `ModelAccuracyWithTimeWindowTest.kt`)